### PR TITLE
Ensure offcanvas is hidden when navigating back

### PR DESCRIPTION
**Pull request by janbaykara**
## Description

Manually re-initialises Offcanvas elements and closes them, so that they never appear to hang open.

## Motivation and Context

This was an issue because Turbo seems to have cached the HTML at the moment of navigation away from the page, with the offcanvas element open.

When one navigates back, it seems Boostrap didn't properly re-initialise the offcanvas — probably because it's not getting the Window events it needs to get going (Turbo gives its own events like `turbo:load` for these fake navigations).

So we help it along.

### CODE REVIEW

**Review by lowlandjuju — CHANGES_REQUESTED** [review]
This is great - much less janky than my previous idea!

I have just one suggestion - for an even better user experience, to modify the reset function to take into account whether a user might have navigated to a path via a link within one of the `Offcanvas` elements. Like this:

```js
function resetJSForTurboFrame() {
  // Ensure Boostrap Offcanvases (sidepanels) are in the correct state
  // according to the Turbo Frame cache'd history. If going back in the history
  // and the Offcanvas was previously open, then re-open it
  const offcanvas = document.querySelectorAll(".offcanvas")

  for (const el of offcanvas) {
    const visible = window.getComputedStyle(el)['visibility'] === 'visible'
    let offcanvas = new Offcanvas(el)
    offcanvas.show()
    if (!visible) offcanvas.hide()
  }
}
```

File: smartforests/typescript/index.tsx [hunk 1]
```diff
@@ -60,3 +62,14 @@ async function load() {
 
   modules.forEach(fn => fn());
 }
+
+function resetJSForTurboFrame() {
```
Comment (lowlandjuju) [inline smartforests/typescript/index.tsx:66]: Suggested change to this in the main review comment

**Comment by lowlandjuju** [comment]
@janbaykara have just pushed that suggested change - so then if you're happy with it we can merge that x

**Comment by janbaykara** [comment]
This is great

**Pull request merged**